src/algorithms/standard/directscaletransform.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2006-2021  Music Technology Group - Universitat Pompeu Fabra
+ *
+ * This file is part of Essentia
+ *
+ * Essentia is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License as published by the Free
+ * Software Foundation (FSF), either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the Affero GNU General Public License
+ * version 3 along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+#include "directscaletransform.h"
+#include "essentiamath.h"
+
+#define FOR(i,l,r) for(int i=l; i<r; i++)
+
+using namespace essentia;
+using namespace standard;
+
+const char* DirectScaleTransform::name = "DirectScaleTransform";
+const char* DirectScaleTransform::category = "Standard";
+const char* DirectScaleTransform::description = 
+DOC("Computes Direct Scale Transform on a given matrix.\n" 
+    "This code was derived from the original DST paper:\n"
+    "[1] Williams, W. J., and E. J. Zalubas. Helicopter transmission fault detection via "
+    "time-frequency, scale and spectral methods. Mechanical systems and signal processing"
+    "14.4 (2000): 545-559.");
+
+void DirectScaleTransform::configure() {
+    _C = parameter("C").toReal();
+    _fs = parameter("fs").toReal();
+}
+
+void DirectScaleTransform::compute() {
+
+    const std::vector<std::vector<Real>>& matrix = _matrix.get();
+    std::vector<std::vector<Real>>& result = _result.get();
+
+    int N = matrix.size();
+    if (N == 0) throw EssentiaException("Input matrix is empty");
+
+    std::complex<Real> zi = std::complex<Real>(0, 1);
+    Real step = M_PI/log(N+1);
+    int num_rows = _C/step+1;       // Number of rows. +1 because need to include the first row which is the 0th multiple of step
+    int P = matrix[0].size();
```

Comment:
There's no guarantee on the consistency of the size of inner vectors, because we are using vector of vectors. Therefore, this should be checked.